### DISCUSSION TITLE
Version Numbers and Branches

### DISCUSSION THREAD

**Discussion by TonyValenti** [oqtane/oqtane.framework#825] [Ideas]
A few ideas I wanted to toss out.

1.  Between .NET Core versions, Microsoft has not been shy about breaking things in order to provide a better over-all platform.  I've personally seen this happen with EFCore and there has always been significant benefit in doing so.  It may be good to consider .NET version upgrades as an acceptable time to make major refactors to the core.  Knowing that you might need to make a few adjustments to your Oqtane code at the same time you're doing a .NET Core upgrade has benefits.

2.  Just as Microsoft is aligning all version numbers against the .NET Core version numbers, it may be a good idea to do with Oqtane.  This helps answer a lot of questions like "What version of .NET is Oqtane x supposed to run on?".  This would mean that the next version of Oqtane should be V5.xx.xx

3.  For source code branching, it would be nice if all enhancements to Oqtane took place against the Main/Master branch such that Main/Master is always V.Next and whenever an official release of Oqtane is made, the source code for that version is branched.  This simplifies development and allows people to easily get source for whatever version they want.  Also, if there is ever a need for a major fix (security patch or such) it can be applied to Main and ported to the appropriate versions.  FYI - I'm not a Git expert so there may be some nuances to this idea.

**Comment by sbwalker** [comment]
1. I was told by a member of the ASP.NET team that their goal was to have no breaking changes in the upgrade from .NET Core 3.1 to .NET 5. They made this decision because they did not want to restrict people from upgrading. Oqtane upgraded from .NET Core 3.1 to .NET 5 without any breaking changes. Perhaps EF Core had a slightly different philosophy - however in general they must have preserved compatibility as Oqtane experienced no upgrade issues in regards to EF Core. The Oqtane rationale for avoiding breaking changes at this stage is documented in the Philosophy. We will not be introducing breaking changes in V2.

2. I am not in favor of incrementing the Oqtane version to V5 to align with .NET 5.  

3. Please see #799 for a whole discussion on branching.
